packages/compiler/core/types.ts

```diff
@@ -281,6 +281,16 @@ export interface Interface extends BaseType, DecoratedType, TemplatedTypeBase {
   node: InterfaceStatementNode;
   namespace?: Namespace;
 
+  /**
+   * The interfaces that provide additional operations via `interface extends`.
+   *
+   * Note that despite the same `extends` keyword in source form, this is a
+   * different semantic relationship than the one from {@link Model} to
+   * {@link Model.baseModel}. Operations from extended interfaces are copied
+   * into {@link Interface.operations}.
+   */
+  sourceInterfaces: Interface[];
```

Comment:
Is this related to this change?

Reply:
Yes, it's also a distinct feature that was on the schedule for next sprint, but I realized I needed it to get the validation working. Once I correctly fixed validation to not fire on template instantiations themselves, I needed a way to still fire on the reference to the instantiation in `interface Foo extends Bar<NewerThanFoo>`. We had similar handling for model instantiations, but nowhere clean to get this reference to interface instantiation so I added the feature now and used it.